### ISSUE TITLE
decodeBigInteger doesn't work properly

### ISSUE THREAD

**Issue by tinchou** [ethereum/ethereumj#917]
Here's a test case:

```
    @Test
    public void encodeDecodeBigInteger() {

        BigInteger expected = new BigInteger("9650128800487972697726795438087510101805200020100629942070155319087371611597658887860952245483247188023303607186148645071838189546969115967896446355306572");
        byte[] encoded = encodeBigInteger(expected);
        BigInteger decoded = decodeBigInteger(encoded, 0);
        assertNotNull(decoded);
        assertEquals(expected, decoded);
    }
```

**Comment by bendyarm** [comment]
I extended my pull request #936  to fix about 29 bugs in RLP including incorporating mlahia's fix.

**Comment by tinchou** [comment]
Thanks for taking a look at this!

I read you fix some 29 bugs, but aren't you adding any tests? It's great to have a correct implementation, but without them you never know when you'll break things again.

**Comment by bendyarm** [comment]
Good point.  I will add tests for some edge cases later next week.

**Comment by zilm13** [comment]
Should be resolved with #975
